src/Dependencies/Collections/SegmentedList`1.cs

```diff
@@ -1198,7 +1198,7 @@ public void Sort(Comparison<T> comparison)
 
             if (_size > 1)
             {
-                SegmentedArray.Sort<T>(_items, 0, _size, Comparer<T>.Create(comparison));
+                SegmentedArray.Sort(_items, 0, _size, comparison);
```

Comment:
existing codepath forced an allocation from comparison to comparer.  That comparer was *later* converted back to a comparison by grabbing off the .Compare method from it.  So this double allocated going through this path.

THe path that takes an IComparer single-allocs (tearing off the .Compare delegate).

Now, there is at least no alloc if passing in a comparison delegate.  we pass it through all the way to the final place that uses it.

Reply:
Allocations caused by this contributed to nearly half of the excess costs here (as well as a ton of garbage): 

![image](https://github.com/dotnet/roslyn/assets/4564579/14d5a3c0-42df-4cec-85a0-19b3bc35ceb4)